### PR TITLE
(PCP-893) Honor CRL when communicating with broker/puppetserver

### PR DESCRIPTION
**Pull request by donoghuc**
This commit updates pxp-agent to use a new required `ssl-crl` configuration option. This option represents the CRL to validate when communicating with PCP brokers or puppet master/server.

### CODE REVIEW

**Comment by donoghuc** [comment]
Started with leatherman https://github.com/puppetlabs/leatherman/pull/320

**Comment by donoghuc** [comment]
cpp-pcp work https://github.com/puppetlabs/cpp-pcp-client/pull/248

File: lib/tests/unit/agent_test.cc [hunk 1]
```diff
@@ -26,6 +26,7 @@ TEST_CASE("Agent::Agent", "[agent]") {
                                                getCaPath(),
                                                getCertPath(),
                                                getKeyPath(),
+                                               getCrlPath(),
```
Comment (MikaelSmith) [inline lib/tests/unit/agent_test.cc]: Not sure how easy it is, but can we add a negative test that shows connecting to a cert listed in the CRL is rejected?

Comment (donoghuc) [reply lib/tests/unit/agent_test.cc]: Added tests for this in cpp-pcp-client. Also added extensive coverage for this in acceptance tests (show that the curl client as well as web-socket connection check CRL and will not connect when a cert has been revoked).

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by McdonaldSeanp** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Comment by donoghuc** [comment]
jenkins please test this on redhat7-64a,windows2016-64a

**Pull request merged**